feature_engine/selection/information_value.py

```diff
@@ -0,0 +1,328 @@
+         The difference between the binomial distributions of positive and
+         negative classes for each unique value of the selected categorical
+         variables.
+
+    woe_encoder_dict_:
+        Contains the weights of evidence (WoE) for all the unique values of
+        the selected categorical variables.
+
+    information_values_:
+        A dictionary comprised of the information values for the selected
+        categorical variables.
+
+    Methods
+    -------
+    fit:
+        Calculates the information values for the selected categorical
+        variables.
+
+    {fit_transform}
+
+    transform:
+        Returns a dataframe with the variables and their corresponding
+        information values.
+
+    See Also
+    --------
+    feature_engine.encoding.WoEEncoder
+    """
+
+    def __init__(
+            self,
+            variables: Union[None, int, str, List[Union[str, int]]] = None,
+            sort_values: bool = False,
+            ignore_format: bool = False,
+            errors: str = "ignore",
+    ) -> None:
+
+        if not isinstance(sort_values, bool):
+            raise ValueError(
+                "sort_values must be a boolean variable. Got {sort_values} instead."
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.sort_values = sort_values
+
+        # parameters are checked when WoEEncoder is instatiated
+        self.ignore_format = ignore_format
+        self.errors = errors
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """
+        Learn the information value.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+            Can be the entire dataframe, not just the categorical variables.
+
+        y: pandas series of shape = [n_samples, ]
+            Target, must be binary.
+
+        """
+        # check input dataframe
+        X, y = check_X_y(X, y)
+
+        if y.nunique() != 2:
+            raise ValueError(
+                "This selector is designed for binary classification. The target "
+                "used has more than 2 unique values."
+            )
+
+        # find categorical variables or check variables entered by user
+        self.variables_ = _find_or_check_categorical_variables(X, self.variables)
+
+        # derive the difference in the binomial distributions for each unique value
+        # for each selected categorical variable
+        self.class_diff_encoder_dict_ = self._calc_diff_between_class_distributions(
+            X, y
+        )
+
+        # get WoE values for unique values of selected categorical variables
+        self.woe_encoder_dict_ = self._calc_woe_encoder_dict(X, y)
+
+        # get information values for unique values of selected categorical variables
+        self.information_values_ = self._calc_information_values()
+
+        self._get_feature_names_in(X)
```

Comment:
as part of fit() we also need to identify the features that will be dropped in an attribute called features_to_drop_

see for example the class SelectByShuffling.